CleverCloud/clever-components
# Storybook (dev server): refresh CEM when a component is updated

Fixes #989 

## What does this PR do?

- Improve the `viteFinal` config handling. We rely on the `mergeConfig` helper from vite to avoid erasing parts of the builder config by mistake. This is on par with the Storybook docs (see [viteFinal - Storybook](https://storybook.js.org/docs/api/main-config-vite-final#options)).
- Add vite to our dev dependencies. We need to for the `mergeConfig` helper (although we could rely on the `storybook` node modules one) + the types. Note that since v8.0.0, storybook automatically adds vite to the dev dependencies so this feels like the right move anyway?
- Regenerate CEM everytime a component is updated (in dev mode only). (see #989 for more info + the comments within the code to review).

## How to review?

- Check both commits (the first commit's diff is not easy to read, you might as well just check the code itself for the config at least),
- Run storybook locally, go to a component's doc page and update the doc of the component you are viewing. The doc story should reload and you should see the modified doc.
- The "built" version (preview/prod) should not be affected by these changes so nothing to check expect that the preview has been built and is working.
